scripts/qa-runtime-daemon-lifecycle.sh

```
@@ -0,0 +1,1067 @@
+        sleep 1
+    done
+
+    if kill -0 "$pid" 2>/dev/null; then
+        record_result "PASS" "runtime_mode_on_demand" \
+            "on-demand mode process alive (idle or ready)" \
+            "pid=$pid"
+    else
+        # Process exited — may be expected for on-demand without models
+        local log_content=""
+        log_content="$(tail -20 "$LOG_DIR/$label.log" 2>/dev/null || true)"
+        if [[ "$log_content" == *"help"* ]] || [[ "$log_command" == *"usage"* ]]; then
```

Shellcheck (0.11.0)

[warning] log_command is referenced but not assigned.

(SC2154)